zebra-network/src/protocol/external/tests/prop.rs

```diff
@@ -227,4 +233,139 @@ proptest! {
             hex::encode(&addr_bytes2),
         );
     }
+
+    /// Test round-trip AddrV2 serialization for gossiped MetaAddrs
+    #[test]
+    fn addr_v2_gossiped_roundtrip(gossiped_addr in MetaAddr::gossiped_strategy()) {
+        zebra_test::init();
+
+        // We require sanitization before serialization
+        let gossiped_addr = gossiped_addr.sanitize();
+        prop_assume!(gossiped_addr.is_some());
+        let gossiped_addr = gossiped_addr.unwrap();
+
+        // Check that malicious peers can't make Zebra's serialization fail
+        let addr_bytes = AddrV2::from(gossiped_addr).zcash_serialize_to_vec();
+        prop_assert!(
+            addr_bytes.is_ok(),
+            "unexpected serialization error: {:?}, addr: {:?}",
+            addr_bytes,
+            gossiped_addr
+        );
+        let addr_bytes = addr_bytes.unwrap();
+
+        // Assume other implementations deserialize like Zebra
+        let deserialized_addr = AddrV2::zcash_deserialize(addr_bytes.as_slice());
+        prop_assert!(
+            deserialized_addr.is_ok(),
+            "unexpected deserialization error: {:?}, addr: {:?}, bytes: {:?}",
+            deserialized_addr,
+            gossiped_addr,
+            hex::encode(addr_bytes),
+        );
+        let deserialized_addr: AddrV2 = deserialized_addr.unwrap();
+        let deserialized_addr: MetaAddr = deserialized_addr.try_into().expect("arbitrary MetaAddrs are IPv4 or IPv6");
+
+        // Check that the addrs are equal
+        prop_assert_eq!(
+            gossiped_addr,
+            deserialized_addr,
+            "unexpected round-trip mismatch with bytes: {:?}",
+            hex::encode(addr_bytes),
+        );
+
+        // Now check that the re-serialized bytes are equal
+        // (`impl PartialEq for MetaAddr` might not match serialization equality)
+        let addr_bytes2 = AddrV2::from(deserialized_addr).zcash_serialize_to_vec();
+        prop_assert!(
+            addr_bytes2.is_ok(),
+            "unexpected serialization error after round-trip: {:?}, original addr: {:?}, bytes: {:?}, deserialized addr: {:?}",
+            addr_bytes2,
+            gossiped_addr,
+            hex::encode(addr_bytes),
+            deserialized_addr,
+        );
+        let addr_bytes2 = addr_bytes2.unwrap();
+
+        prop_assert_eq!(
+            &addr_bytes,
+            &addr_bytes2,
+            "unexpected round-trip bytes mismatch: original addr: {:?}, bytes: {:?}, deserialized addr: {:?}, bytes: {:?}",
+            gossiped_addr,
+            hex::encode(&addr_bytes),
+            deserialized_addr,
+            hex::encode(&addr_bytes2),
+        );
+    }
+
+    /// Test round-trip AddrV2 serialization for all MetaAddr variants after sanitization
+    #[test]
+    fn addr_v2_sanitized_roundtrip(addr in any::<MetaAddr>()) {
```

Comment:
This test seems like a superset of the previous test. Is there a reason to keep both?

Reply:
They used to be different, but they're the same now.

Deleted the redundant tests in bba2dade6.